src/source.rs

```diff
@@ -463,6 +470,54 @@ pub fn new_http_client() -> Result<blocking::Client> {
         .context("building HTTP client")
 }
 
+/// Wrapper around Client::get() with error handling based on HTTP return code and optionally basic
+/// exponential backoff retries for transient errors.
+pub fn http_get(
+    client: blocking::Client,
+    url: &str,
+    retries: Option<FetchRetries>,
+) -> Result<blocking::Response> {
+    // this matches `curl --retry` semantics -- see list in `curl(1)`
+    const RETRY_STATUS_CODES: [u16; 6] = [408, 429, 500, 502, 503, 504];
+
+    let mut delay = 1;
+    let (infinite, mut tries) = match retries {
+        Some(FetchRetries::Infinite) => (true, 0),
+        Some(FetchRetries::Finite(n)) => (false, n.get() + 1),
+        None => (false, 1),
```

Comment:
This is okay, but would it make more sense to drop the containing `Option<>` and add a `None` case to `FetchRetries`?

Reply:
Yeah, debated this a bit at the beginning, though with the switch to NonZeroU32 typing I think it makes sense.